### ISSUE TITLE
[Bug]: Images uploaded for notifications expire after 1h

### ISSUE THREAD

**Issue by pjeweb** [alveusgg/alveusgg#771]
### What is the current behavior?

The notification file upload is missing the call to `checkAndFixUploadedImageFileStorageObject` so it gets garbage collected after 1h.

### What is the expected behavior?

Valid uploads stay available at least as long as the notification is active or forever. Invalid uploads should show an error (non-image files).

### Steps to reproduce

Go to `/admin/notifications`, upload file, send notification with image, check the `FileStorageObject` in db expires within 1h.

### Browser specs

n/a

**Comment by git-divyansh** [comment]
I would like to take up this issue!

**Comment by continentaldivide** [comment]
Hi folks, found this project via watching some Alveus streams.  Thanks for the contributor-friendly efforts you've made here w/r/t great documentation and helpful issue tagging.

A lot of what you've built is a bit over my head for now, but I'm interesting in poking at some of the low-hanging fruit to see whether it's within my reach to solve/contribute to -- possibly starting with this one.

I'm running into a problem early in investigating this issue and wanted to ask whether you could steer me in the right direction.  I've got the app configured and running locally.  I've added `DISABLE_ADMIN_AUTH=true` to `apps/website/.env` in an attempt to skip over dealing with any kind of Twitch integration for now.  My expectation was that this would allow me to navigate to `/admin/notifications` and start to get some context for the described issue.  Instead, I'm redirected to `/auth/signin` and an error is displayed:
```
Error signing in

Unable to sign in.
```
Is this expected behavior / have I misunderstood the documentation?  I think I can dig into the Twitch integration and figure out how to authorize myself that way, but wanted to also correct whatever wrong turn I've taken on this piece as well.

Thanks!

**Comment by MattIPv4** [comment]
Hm, I'm not able to reproduce that. I'm setting `DISABLE_ADMIN_AUTH=true` in `apps/website/.env` and then restarting the application, and can then access `http://localhost:3000/admin/notifications` without issue while showing as signed out in the nav.

@brianpeiris might have more ideas as they just implemented this flag?

**Comment by continentaldivide** [comment]
Appreciate you double-checking -- it's helpful to have confirmation that it works as expected on someone else's machine.  I'm sure you're right that going the Twitch route is going to be the better move anyway...I'll give that a shot next time I take a look at it.  Thanks!

**Comment by brianpeiris** [comment]
I wasn't able to reproduce the admin issue with a fresh clone. Here's my diff between `.env` and `.env.example`, if it helps. You'll want to make sure these other secrets and keys are configured, in addition to setting `DISABLE_ADMIN_AUTH=true`

```diff
--- .env.example        2025-01-21 19:07:01.107971549 -0500
+++ .env        2025-01-21 19:10:37.062748299 -0500
@@ -32,7 +32,7 @@ SHADOW_DATABASE_URL='mysql://root:alveus
 # Next Auth
 #   You can generate the secret via 'openssl rand -base64 32' on Linux
 # More info: https://next-auth.js.org/configuration/options#secret
-NEXTAUTH_SECRET=
+NEXTAUTH_SECRET=[value from openssl rand -base64 32]
 NEXTAUTH_URL=http://localhost:3000

 # Twitch OAuth
@@ -57,11 +57,11 @@ DISCORD_CHANNEL_WEBHOOK_URLS_ANNOUNCEMEN

 # Action API secret
 #   You can generate the secret via 'openssl rand -base64 32' on Linux
-ACTION_API_SECRET=
+ACTION_API_SECRET=[value from openssl rand -base64 32]

 # Vercel Cron secret (see https://vercel.com/docs/cron-jobs/manage-cron-jobs#securing-cron-jobs)
 #   You can generate the secret via 'openssl rand -base64 32' on Linux
-CRON_SECRET=
+CRON_SECRET=[value from openssl rand -base64 32]

 # Weather API secret
 #   You can add a weather station to wunderground.com for this
@@ -72,8 +72,8 @@ WEATHER_STATION_ID=INUNAV15

 # Web Push VAPID Key
 #   You can generate the keys via 'npx web-push generate-vapid-keys'
-NEXT_PUBLIC_WEB_PUSH_VAPID_PUBLIC_KEY=
-WEB_PUSH_VAPID_PRIVATE_KEY=
+NEXT_PUBLIC_WEB_PUSH_VAPID_PUBLIC_KEY=[value from pnpx web-push generate-vapid-keys]
+WEB_PUSH_VAPID_PRIVATE_KEY=[value from pnpx web-push generate-vapid-keys]
 WEB_PUSH_VAPID_SUBJECT=mailto:admin@alveus.gg

 # File Storage (S3-compatible)
@@ -90,14 +90,14 @@ UPSTASH_QSTASH_URL=https://qstash.upstas
 UPSTASH_QSTASH_KEY=

 # Disable admin authentication for local development
-DISABLE_ADMIN_AUTH=false
+DISABLE_ADMIN_AUTH=true

 # Super users
 SUPER_USER_IDS=

 # Encryption passphrase for data at rest
 #   You can generate the secret via 'openssl rand -base64 24' on Linux
-DATA_ENCRYPTION_PASSPHRASE=
+DATA_ENCRYPTION_PASSPHRASE=[value from openssl rand -base64 24]

 # Alveus community photos API (Immich)
 COMMUNITY_PHOTOS_URL=https://upload.alveuscommunity.photos
```

**Comment by continentaldivide** [comment]
Oh wow -- thanks @brianpeiris for the quick and thorough response!  This made the issue clear immediately.  I'd forked the repo about a week ago but a few days went by before I actually started getting everything up and running -- my fork was just before that feature was added, but I'd still been following the up-to-date README.  I'll sync my fork and try again.  🙂

**Comment by fsesma** [comment]
Is this still an issue?

**Comment by MattIPv4** [comment]
I believe so looking at the usages of `checkAndFixUploadedImageFileStorageObject`, @pjeweb?

**Comment by pjeweb** [comment]
> I believe so looking at the usages of `checkAndFixUploadedImageFileStorageObject`, @pjeweb?

Yes! This is still not fixed, but we don't really use the feature

**Comment by pjeweb** [comment]
> I believe so looking at the usages of `checkAndFixUploadedImageFileStorageObject`, @pjeweb?

Yes! This is still not fixed, but we don't really use the feature

**Comment by fsesma** [comment]
Gotcha! Should the corresponding code be removed if the feature is no longer in use?
